pkg/controller/networkpolicy/controller_query_test.go

```diff
@@ -0,0 +1,216 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy
+
+import (
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+// pods represent kubernetes pods for testing proper query results
+var pods = []v1.Pod{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "podA",
+			Namespace: "nsA",
+			Labels:    map[string]string{"group": "appliedTo"},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name: "container-1",
+			}},
+			NodeName: "nodeA",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+			PodIP: "1.2.3.4",
+		},
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "podB",
+			Namespace: "nsB",
+			Labels:    map[string]string{"group": "appliedTo"},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name: "container-1",
+			}},
+			NodeName: "nodeA",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+			PodIP: "1.2.3.4",
+		},
+	},
+
+}
+
+// polices represent kubernetes policies for testing proper query results
+//
+// policy 0: select all pods and deny default ingress
+// policy 1: select all pods and deny default egress
+
+var policies = []networkingv1.NetworkPolicy{
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default-deny-ingress",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeIngress,
+			},
+		},
+	},
+	{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: 		"default-deny-egress",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+		},
+	},
+}
+// TestInvalidSelector tests the result of QueryNetworkPolicy when the selector (right now pod, namespace) does not
+// select any pods
+func TestInvalidSelector(t *testing.T) {
+	// create controller
+	_, controller := newController()
+	// create querier with stores inside controller
+	address, appliedToGroup, policy := controller.addressGroupStore, controller.appliedToGroupStore, controller.internalNetworkPolicyStore
+	endpointQuerier := EndpointQueryReplier{address, appliedToGroup, policy}
+	// test appropriate response to QueryNetworkPolices
+	namespace, pod := "non-existing-namespace", "non-existing-pod"
+	applied, egress, ingress := endpointQuerier.QueryNetworkPolicies(namespace, pod)
+	if applied != nil || egress != nil || ingress != nil {
```

Comment:
we use https://github.com/stretchr/testify for test assertions. Please refer to the doc and other unit tests in Antrea.

Reply:
will do